src/pathogena/batch_upload_apis.py

```diff
@@ -0,0 +1,197 @@
+from typing import Any
+
+import requests
+
+
+class APIError(Exception):
+    """Custom exception for API errors."""
+
+    def __init__(self, message: str, status_code: int):
+        super().__init__(message)
+        self.status_code = status_code
+
+
+class APIClient:
```

Comment:
i couldn't test this locally but i also couldn't see the reference in code we initialise this client.